### PR TITLE
ACCUMULO-4597 fixed bug in rfile-info

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java [hunk 1]
```diff
@@ -1015,14 +1018,23 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
...
       Map<String,ArrayList<ByteSequence>> cf = new HashMap<>();
 
       for (LocalityGroupMetadata lcg : localityGroups) {
-        ArrayList<ByteSequence> setCF = new ArrayList<>();
+        ArrayList<ByteSequence> setCF;
 
-        for (Entry<ByteSequence,MutableLong> entry : lcg.columnFamilies.entrySet()) {
-          setCF.add(entry.getKey());
+        if (lcg.columnFamilies == null) {
+          Preconditions.checkState(lcg.isDefaultLG, " Group %s has null families. Only expect default locality group to have null families.", lcg.name);
```
Comment (phrocker) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java]: Is there any other useful information that could be provided? For example, would it make sense to print out the rest of the locality group column families and help identify if others are listed as a the default and have no column families?

Comment (phrocker) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java]: haha and it was merged!

Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java]: You can thank findbugs for the message.  I had some different code that accomplished the same thing and made findbugs unhappy.  So I changed it to this to soothe findbugs.  I don't expect this case to happen, and if it does I would have been happy with an NPE.  There are many places that could have an NPE when there is a bug, we don't handle them all with detailed error messages.

Findbugs has found some nasty bugs.   The false positives can be a little annoying, but I think its worth it.

Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java]: > Findbugs has found some nasty bugs. The false positives can be a little annoying, but I think its worth it.

+1000

Comment (phrocker) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java]: "There are many places that could have an NPE when there is a bug, we don't handle them all with detailed error messages." 
https://s-media-cache-ak0.pinimg.com/736x/88/15/5e/88155efa961948d0627e61236c7d1f2b.jpg

Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java]: @phrocker did you clear that meme with @mjwall

Comment (mjwall) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java]: +1

**Pull request merged**